Account health calculator

saml33 opened the pull request.

saml33 committed:
create calculator page

saml33 committed:
page layout

ImpossiblePairs committed:
Link Input Fields and Formuals

Linked up the inputs and added the calculations.

State objects/functions are not optimized, as they were built one by one to make sure each was working. Once all components are working, this should be refined.

ImpossiblePairs committed:
Slider Fixes, Additional Details

Fixed slider to manipulate all prices equally instead of collateral.

Added additional information to scenario details: Min Collateral Ratio, Risk rating, and Percent move to liquidation.

ImpossiblePairs committed:
Negative number handling

Updated details to zero out negative equity or leverage, as neither should occur.

saml33 committed:
Merge pull request #42 from ImpossiblePairs/patch-1

Link Input Fields and Formuals

ImpossiblePairs committed:
Update liquidation-calculator.tsx

- Updated handling of state by asset instead of action (BTC.deposit instead of deposit.BTC)
- Fixed slider percentage lag (was one state change behind)
- Cleaned up handling of scenario liquidation
- Handling of some dust borrows/NaN calculations found in testing

ImpossiblePairs committed:
Handling User Input (decimal / sign first)

Delaying update of state until a number is present (allows decimal place or sign to be entered).

Convert user inputs into absolute values to avoid unwanted calculator output (such as a double negative).

saml33 committed:
Merge pull request #45 from ImpossiblePairs/patch-1

Updates from feedback

ImpossiblePairs committed:
Added Calculator For Disconnected State

- Added ability to use liquidation calculator without requiring user to connect a wallet.
- Fixed rounding of prices during slider manipulation and out of focus.
- Reset slider position when price inputs are in focus (erroneous calculations when manually inputting price with slider at percentages other than 100%)

saml33 committed:
Merge pull request #46 from ImpossiblePairs/patch-1

Added Calculator For Disconnected State & bug fixes

saml33 committed:
responsive, layout tweaks and link from trade page

ImpossiblePairs committed:
Account Switching Reload & Leverage Rounding Fix

Switching margin accounts now triggers the Risk Calculator to reload balances.

Rounding method for calculations standardized to floorToDecimal to remove borrow rounding errors causing incorrect leverage calculations.

saml33 committed:
Merge pull request #48 from ImpossiblePairs/account-health-calculator

Account Switching Reload & Leverage Rounding Fix

saml33 committed:
fix typo

dboures commented:
Hey, just a quick few things I noticed when I was using the calculator:

1. If you move the “Edit All Prices Slider” and then click into a price input box, it resets all of the prices. I'm not sure if users will find it confusing but I was not expecting that behavior.
2. Sometimes the decimal precision displayed changes in the price column (For example, click back and forth between the borrows and price columns)
3. Weird UI stuff happens if someone enters too many digits into a calculator cell. Maybe we impose a maximum on the inputs?

ImpossiblePairs committed:
Update Slider & Price Input Behavior

Updated slider & price input behavior after receiving some feedback from Sam and Dboures about price input field and slider interaction.

Originally clicking on the price input would reset the slider and return the prices in the input form back to the reference prices held in state, Now when the slider is in use, users can click on a price input and the price and slider will remain in the same position. If a change is made, the reference prices of all assets will be updated to their currently displayed value and the slider will be reset. If no changes are made, the reference prices remain unchanged and the slider remains at it's current setting.

saml33 committed:
Merge pull request #49 from ImpossiblePairs/account-health-calculator

Update Slider & Price Input Behavior

dboures committed:
Update collateral ratio rounding & liquidation message criteria

saml33 committed:
Merge pull request #50 from dboures/account-health-calculator

Update collateral ratio rounding & liquidation message criteria
